### PR TITLE
simplify implementation of contained_levels, at the cost of more queries

### PR DESCRIPTION
**Pull request by Hamms**
## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: dashboard/test/integration/db_query_test.rb [hunk 1]
```diff
@@ -20,7 +20,7 @@ def setup
       level: level,
       level_source: create(:level_source, level: level)
 
-    assert_cached_queries(18) do
+    assert_cached_queries(25) do
```
Comment (Hamms) [inline dashboard/test/integration/db_query_test.rb:23]: So after a bit of tweaking, this is the only remaining query count that increases with this change.

Note, however, that this test was recently increased from 13 to 18 in https://github.com/code-dot-org/code-dot-org/pull/41420/files#diff-e5cda77c673c69979f0acdd7d326c708c2c051f612c80ebfdcf072fb318d172a, so this jump to 25 represents a near-doubling of the original query count. It's not clear to me how much of a concern that is.

Comment (Hamms) [reply dashboard/test/integration/db_query_test.rb:23]: @code-dot-org/infrastructure any thoughts on this query increase?

Comment (sureshc) [reply dashboard/test/integration/db_query_test.rb:23]: We can use NewRelic, AWS Performance Insights, and the Percona Monitoring “Query Analytics” tool to assess the impact of the current query and predict the impact of this proposed query change. Setup 15 minutes with me tomorrow to review how to do this?

**Review by davidsbailey — APPROVED** [review]
My $0.02 is that this should be fine for performance because there are not typically contained levels in hoc-scale scripts. Also, I believe we have other query-count tests covering the hoc scale scenarios we care about, such as https://github.com/code-dot-org/code-dot-org/blob/a1d23a954f0bea08bb677544fa7bf6e3328ab226/dashboard/test/integration/caching_test.rb#L3

File: dashboard/test/integration/db_query_test.rb (hunk 1 above)
Comment (Hamms) [reply dashboard/test/integration/db_query_test.rb:23]: That'd be great, thanks! I'd love to learn a bit about those tools in general

**Pull request closed without merging**